### PR TITLE
Upgrade to Go 1.11 and use modules

### PR DESCRIPTION
**Pull request by abhinav**
This modernizes the repository organization a bit. This,

- Upgrades to Go 1.11.
- Uses Go modules instead of Glide.
- Drops errcheck, which we rarely use in our codebase internally.
- Uses `./...` to run build, tests, etc. since that does not have to
  account for vendor anymore.
- Drops `-installsuffix cgo` from the production build. This was a
  workaround for pre-1.10 versions of Go. See
  https://github.com/golang/go/issues/9344#issuecomment-69944514.
- Uses a tools.go as recommended in [go-modules-by-example/tools][1] to
  pin to tool dependencies.

[1]: https://github.com/go-modules-by-example/index/tree/master/010_tools

Tested locally with `make docker-launch`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,61 +1,37 @@
...
-errcheck:
-	go install ./vendor/github.com/kisielk/errcheck
-	errcheck $(PKGS)
+	go vet ./...
 
 .PHONY: staticcheck
 staticcheck:
-	go install ./vendor/honnef.co/go/tools/cmd/staticcheck
-	staticcheck $(PKGS)
+	go install honnef.co/go/tools/cmd/staticcheck
```
Comment (glibsm) [inline Makefile]: `GO111MODULE` makes this module aware, right?

Comment (abhinav) [reply Makefile]: Yep. With the env var set, it'll pick up the one we're pinned to.

Comment (glibsm) [reply Makefile]: I'm more curious about `command -v staticcheck` in this case, would it also be somewhere scoped just to this path or global?

Comment (abhinav) [reply Makefile]: No, the command is still installed globally. An option here is to do `go build -o $somepath $importPath` and that will leave the global version unchanged.

Comment (glibsm) [reply Makefile]: Not a huge deal, I'm fine with either. I think second is slightly better practice but it's not any different than it was before

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,19 @@
+module go.uber.org/sally
+
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/golang/lint v0.0.0-20181217174547-8f45f776aaf1
+	github.com/julienschmidt/httprouter v1.2.0
+	github.com/kisielk/gotool v1.0.0 // indirect
+	github.com/kr/pretty v0.1.0 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/stretchr/objx v0.1.1 // indirect
```
[the remark this answers is not in the document]
Comment (abhinav) [reply go.mod]: Oh, weird. Fixing.

File: Makefile (hunk 1 above)
Comment (abhinav) [reply Makefile]: Oh good call. I'll switch.

**Pull request merged**